### ISSUE TITLE
Seneca do not convert mongoose object to JSON

### ISSUE THREAD

**Issue by longweiquan** [senecajs/seneca#281]
Hi seneca team,

I use mongoose with seneca in my project. I hope that seneca can handle mongoose object. Actually, when we use seneca respond as a mongoose callback, seneca will return a timeout error. 

```
seneca.act('role:users, cmd:get', (msg, respond) => {
    UserMongooseModel.findById(msg.id, respond); // seneca timeout exception
});
``` 

But if we transform mongoose object to JSON, it works.

```
seneca.act('role:users, cmd:get', (msg, respond) => {
     UserMongooseModel.findById(msg.id, (error, user) => {
            respond(error, user && user.toJSON());
     }); 
});
``` 

So if possible, I hope seneca can do the JSON conversion at transport layer. Thank you in advance.

**Comment by mcdonnelldean** [comment]
@longweiquan What transport are you using in this instance? Just the built in one?

**Comment by longweiquan** [comment]
Hello,

@mcdonnelldean, yes, just the built-in one with tcp protocol. And I'm on version 0.9.2.

**Comment by mcdonnelldean** [comment]
Cool, leave it with me, I'll ask what should be done here.

**Comment by mcollina** [comment]
Mongoose objects are not serializable to JSON as they are, and `obj.toJSON()` needs to be called. The question here is if we should automatically call `response.toJSON()` if such method is available. I'm for doing nothing here, but I do not like mongoose at all.

**Comment by mcdonnelldean** [comment]
@mcollina Yeah I don't like the idea of calling .toJSON() as it's specific to Mongoose. Having said that, custom serialiser hooks might not be a bad idea, that way they can be loose without additional issues.

**Comment by longweiquan** [comment]
@mcdonnelldean , @mcollina , Yeah a custom serialiser hook is a good idea. :+1:

**Comment by mcdonnelldean** [comment]
@longweiquan Awesome, I'll add an issue to get discussions on this. There is some high priority stuff going on first so it will be a little while before we even get to discuss it but I have noted it down.

**Comment by rjrodger** [comment]
The new Seneca-entity plugin will need to provide custom serializers too :)

So Seneca needs to define a way for plugins to do this.

Currently it's a hack in Seneca-transport 

One big issue: full traversal of each message is bad bad bad for performance, so let's limit this to top level object or properties only (as currently the case)

**Comment by jeltok** [comment]
I use Mongoose with Seneca as well, and I am using 'lean()' function of Mongoose. That works just great.

`        var query = Stats.find()
            .where('date', date)
            .where('eventType').in(events);

        query.lean().exec(...);`

**Comment by mcdonnelldean** [comment]
@rjrodger @mcollina @davidmarkclements This may be important as it deals with transports. Merged with #397
